pandas/core/reshape/merge.py

```diff
@@ -753,6 +744,17 @@ def _maybe_drop_cross_column(
         if cross_col is not None:
             del result[cross_col]
 
+    @cache_readonly
```

Comment:
Aside, maybe we can use 

```
from functools import cache (PY3.9)/ lru_cache

@property
@lru_cache(maxsize=None)/@cache
def _indicator_name(...)
```

in the future instead

Reply:
https://docs.python.org/3/library/functools.html#functools.cached_property

Reply:
To replace `cache_readonly` in general. Could gain some nice benefits like cache clearing and thread safety (if I'm viewing the implementation correctly)